### DISCUSSION TITLE
React Hook form Controller with Material ui Autocomplete not working

### DISCUSSION THREAD

**Discussion by giordifungula** [react-hook-form/react-hook-form#3365] [Q&A] [answered]
## I have a question around using Controller from react-hook-form and material-ui autocomplete component. 
@bluebill1049 Hi Bill, can you please have a look when you get a second.

###Issue:

- I am trying to use the autocomplete component with react-hook-form Controller Component, but whenever I select a option from the dropdown menu and log the value from reack-hoom-form watch function, I get 0 being logged even though the options are strings.

### What I have tried
- I have removed the controller component that was controlling the autocomplete and tried to set local state variable based on autocomplete onchange value and this works fine and is able to set local state, but I would like to wrap my autocomplete inside a controller. See example of autocomplete below without controller wrapping it.
```
 <Autocomplete
          options={top100Films}
          getOptionLabel={(option) => option.title}
          onChange={(e, data) => {
            setInputValue(data.year);
          }}
          renderInput={(params) => {
            return (
              <TextField
                {...params}
                // inputRef={register}
                label={"Resolution Code"}
                variant="outlined"
                name={"resolutionCode"}
                defaultValue={1993}
                fullWidth
                value={params}
              />
            );
          }}
        />
```
- Below is how I am wrapping my autocomplete component inside a controller component ( this is the intended one ), However when I wrapped it inside a controller component, I attempted to set the value of the form with the onchange handler inside Autocomplete form and the value is still set to 0.
- See example of code below
```
 <Controller
          name="country1"
          as={
            <Autocomplete
              options={options}
              getOptionLabel={(option) => option.name}
              onChange={(e) => setValue("country1", e.target.value)} # setting value of country1 form
              getOptionSelected={(option, value) => option.name === value.name}
              renderInput={(params) => (
                <TextField {...params} label="Country 1" />
              )}
            />
          }
          defaultValue=""
          control={control}
        />
```
- I need to have it the autcomplete inside a controller and would like to get the selected value of the autocomplete menu, but its being set to 0 instead of the actual selected string.
- I have re-created the issue and have it on codesand box see link below.

Thank you in advance!

**Comment by bluebill1049** [comment]
any chance you could upgrade to v6? here is a working example.

https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r?file=/src/MuiAutoComplete.js

here is old version: 

https://codesandbox.io/s/react-hook-form-controller-v5-ii1vs?file=/src/MuiAutoComplete.jsx

**Comment by Achilles0509** [comment, marked as answer]
https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r?file=/src/MuiAutoComplete.js

Please refer this link.
You should overwrite onChange function.

**Reply by bluebill1049** [reply]
thanks for sharing

**Reply by hectormoba** [reply]
I'm using v7, but it's easy to handle as well. You should only call `onChange` in fields object.  Thanks for the example, btw!

**Reply by bryanprimus** [reply]
how about autocomplete with `multiple` props passed?

**Reply by hyalen-coursekey** [reply]
Yeah, it works great! Thanks for sharing. But I'm also wondering what should be the implementation changes when the `multiple` prop is set..
